auth_backends/backends.py

```diff
@@ -215,3 +238,86 @@ def _to_language(locale):
         https://docs.djangoproject.com/en/1.6/topics/i18n/#term-translation-string
     """
     return locale.replace('_', '-').lower()
+
+
+class EdXOAuth2(EdXBackendMixin, BaseOAuth2):
+    name = 'edx-oauth2'
+
+    def endpoint(self):
+        return self.setting('ENDPOINT').strip('/')
+
+    def provider_configuration(self):
+        cache_key = 'edx_oauth2_provider_configuration'
+        config = cache.get(cache_key)
+
+        if not config:
+            config = self.get_json(self.endpoint() + '/.well-known/openid-configuration')
+
+            # Cache for one week since the configuration rarely changes
+            cache.set(cache_key, config, self.setting('PROVIDER_CONFIGURATION_CACHE_TTL', 604800))
```

Comment:
The comment above accurately explains the TTL is one week.

This will use whatever cache is available. What strange results are you imagining for per-worker caches?